pauliopt/pauli/pauli_polynomial.py

```diff
@@ -0,0 +1,68 @@
+from .clifford_gates import CliffordGate
+from .pauli_gadget import PauliGadget
+
+from qiskit import QuantumCircuit
+
+from pauliopt.topologies import Topology
+
+
+class PauliPolynomial:
+    def __init__(self):
+        self.pauli_gadgets = []
+
+    def __irshift__(self, gadget: PauliGadget):
+        self.pauli_gadgets.append(gadget)
+        return self
+
+    def __rshift__(self, pauli_polynomial):
+        for gadget in pauli_polynomial.pauli_gadgets:
+            self.pauli_gadgets.append(gadget)
+        return self
+
+    def __repr__(self):
+        rep_ = ""
+        for gadet in self.pauli_gadgets:
+            rep_ += str(gadet) + "\n"
+        return rep_
+
+    def __len__(self):
+        return self.size
+
+    @property
+    def size(self):
+        return len(self.pauli_gadgets)
```

Comment:
```suggestion
    def __len__(self):
        return len(self.pauli_gadgets)
```

Reply:
Perhaps `__len__` is enough. Let's keep things as consistent as `PhaseCircuit` as possible.